From 78cc39f26ab7d362414802803b6065f351668252 Mon Sep 17 00:00:00 2001
From: Dominic Hargreaves <dom@earth.li>
Date: Sun, 24 May 2015 14:06:05 +0200
Subject: [PATCH] Updates to Perl policy to reflect current status with perl
 5.22

The change to versioned directories is because multiple versions of
libperl (and with it the standard library) are now coinstallable.

Acked-by: Niko Tyni <ntyni@debian.org>
---
 perl-policy.sgml | 25 ++++++++++++-------------
 1 file changed, 12 insertions(+), 13 deletions(-)

diff --git a/perl-policy.sgml b/perl-policy.sgml
index 12cd82c..2572962 100644
--- a/perl-policy.sgml
+++ b/perl-policy.sgml
@@ -129,16 +129,17 @@
       <sect id="paths">
 	<heading>Module Path</heading>
 	<p>
-	  Perl searches three different locations for modules, referred
+	  Perl searches several different locations for modules, referred
 	  to in this document as <var>core</var> in which modules
 	  distributed with Perl are installed, <var>vendor</var> for
 	  packaged modules and <var>site</var> for modules installed by
 	  the local administrator.
 	</p>
 	<p>
-	  The module search path (<tt>@INC</tt>) in the Debian packages
+	  The module search path (<tt>@INC</tt>) in the current Debian packages
 	  has been ordered to include these locations in the following
-	  order:
+	  order<footnote>@INC contains other paths which should be considered
+      internal to the implementation of the perl packaging</footnote>
 	  <taglist>
 	    <tag><var>site</var> (current)</tag>
 	    <item>
@@ -146,7 +147,7 @@
 		Modules installed by the local administrator for the
 		current version of Perl (see <ref id="site">).
 	        <example>
-$Config{sitearch}  (currently /usr/local/lib/perl/<var>version</var>)
+$Config{sitearch}  (currently /usr/local/lib/<var>arch-triplet</var>/perl/<var>version</var>)
 $Config{sitelib}   (currently /usr/local/share/perl/<var>version</var>)
 		</example>
 		Where <var>version</var> indicates the current Perl
@@ -170,9 +171,13 @@ $Config{sitelib}   (currently /usr/local/share/perl/<var>version</var>)
 	      <p>
 		Packaged modules (see <ref id="module_packages">).
 	        <example>
-$Config{vendorarch} (currently /usr/lib/perl5)
+$Config{vendorarch} (currently /usr/lib/<ver>arch-triplet</ver>/perl5/<ver>shortversion</ver>)
 $Config{vendorlib}  (currently /usr/share/perl5)
 	        </example>
+		Where <var>shortversion</var> indicates the current Perl major
+		version (for example <tt>5.22</tt>).
+            </p>
+            <p>
 		These locations, particularly
 		<tt>$Config{vendorarch}</tt>, may change if
 		necessary<footnote>For example, to include
@@ -189,11 +194,11 @@ $Config{vendorlib}  (currently /usr/share/perl5)
 	      <p>
 		Modules included in the core Perl distribution.
 	        <example>
-$Config{archlib} (currently /usr/lib/perl/<var>shortversion</var>)
+$Config{archlib} (currently /usr/lib/<var>arch-triplet</var>/perl/<var>shortversion</var>)
 $Config{privlib} (currently /usr/share/perl/<var>shortversion</var>)
 	        </example>
 		Where <var>shortversion</var> indicates the current Perl major
-		version (for example <tt>5.18</tt>).
+		version (for example <tt>5.22</tt>).
 	      </p>
 	      <p>
 		These locations should be considered internal to the <package>
@@ -300,12 +305,6 @@ make install
 	  described in <ref id="paths"> as <var>vendor</var>.
 	</p>
 	<p>
-	  No version subdirectory exists on these directories as the
-	  dependencies for packaged modules (see <ref id="module_deps">)
-	  should ensure that all work with the current
-	  <package>perl</package> package.
-	</p>
-	<p>
 	  The Perl distribution includes many modules available
 	  separately from CPAN<footnote><url
 	  id="http://www.perl.com/CPAN"></footnote>, which may have a
-- 
2.1.4

